utils/lobby-data.ts

```diff
@@ -3,14 +3,33 @@ export type ILobbyCreationData = {
   readonly id: string;
 }
 
-export type ILobbiesData = {
-  readonly lobbies: string[];
+export type ILobbySong = {
+  id: string;
+  artists: string[];
+  name: string;
+  contributors: string[];
+}
+
+export type ILobbyUser = {
+  id: string,
+  name: string,
 }
 
-export type ILobbyData = {
+export type ILobbyDataShort = {
   readonly id: string;
-  readonly managerId: string;
   readonly name: string;
-  readonly participants: string[];
   readonly theme: string;
+}
+
+export type ILobbiesData = {
+  readonly lobbies: ILobbyDataShort[];
+}
+
+export type ILobbyDataFull = {
+  readonly name: string;
+  readonly theme: string;
+  readonly managerId: string;
+  readonly managerName: string;
+  readonly songs: ILobbySong[];
+  readonly users: ILobbyUser[];
```

Comment:
to me, it seems that we can condense some of this using some clever `interfaces`

```ts
export interface ILobbyDataBase {
  readonly id: string;
  readonly name: string;
}

export interface ILobbyUser extends ILobbyBase {}

export interface ILobbySong extends ILobbyBase {
  readonly artists: string[];
  readonly contributors: string[];
}

export interface ILobbyDataBase extends ILobbyBase {
  readonly theme: string;
}

export interface ILobbyData extends ILobbyDataBase {
  reaodnly managerId: string;
  readonly managerName: string;
  readonly songs: ILobbySong[];
  readonly user: ILobbyUser[];
}

```